nebu/cli/get.py

```diff
@@ -158,88 +181,200 @@ def store_sha1(sha1, write_dir, filename):
+                                              write_dir,
+                                              content_filename,
+                                              legacy_id)
+            tasks.append(asyncio.ensure_future(content_coro))
+
+        def enqueue_extras():
+            if not get_resources:
+                return
+            if 'extras' not in resource_groups:
+                return
+
+            for filename, meta in resource_groups['extras'].items():
+                resource_id = meta['id']
+                resource_url = f'{base_url}/resources/{resource_id}'
+                resource_coro = fetch_resource_node(session,
+                                                    resource_url,
+                                                    write_dir,
+                                                    filename,
+                                                    resource_id)
+                tasks.append(asyncio.ensure_future(resource_coro))
+
+        def enqueue_children():
+            def is_preface(node):
+                return ("Preface" in node['title'])
+
+            if 'contents' not in node:
+                return
+
+            no_bump_index = is_preface(node['contents'][0])
+            for index, child in enumerate(node['contents']):
+                content_meta_url = f'{base_url}/contents/{child["id"]}'
+                content_meta_coro = fetch_content_meta_node(session,
+                                                            child,
+                                                            content_meta_url,
+                                                            write_dir,
+                                                            (index
+                                                             if no_bump_index
+                                                             else index + 1))
+                tasks.append(asyncio.ensure_future(content_meta_coro))
+
+        def get_content_filename():
+            try:
+                return filename_by_type[metadata['mediaType']]
+            except (TypeError, KeyError):
+                return None
+
+        def get_legacy_id():
+            try:
+                return metadata['legacy_id']
+            except (TypeError, KeyError):
+                return None
+
+        def get_scoped_directory():
+            is_module = content_filename == 'index.cnxml'
+            is_collection = content_filename == 'collection.xml'
+
+            node_title = node.get('title') or ''
+            index_string = ('{:02d} '.format(index_in_group)
+                            if not is_collection
+                            else '')
+            tree_dirname = f'{index_string}{_safe_name(node_title)}'
+
+            if is_module and not book_tree:
+                return legacy_id
+            if book_tree:
+                return tree_dirname
+            return ''
+
+        tasks = []
+        metadata = await get_metadata()
```

Comment:
It took me a while to realize that each of these variables (up to `legacy_id`) could be `None` depending on how much information is in the response.

Reply:
Would appending `_or_none` to the end of the function names help?

Reply:
I am not sure. I think that having the variables declared outside the function was the bigger confusing part. However, I'm having trouble thinking of a nicer way to do this so maybe ignore me 😄

Reply:
I made metadata return empty dict instead of None to make it easier to inline those functions and so it should be more evident they can be `None` now.